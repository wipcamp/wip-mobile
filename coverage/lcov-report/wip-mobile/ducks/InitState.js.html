<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for wip-mobile/ducks/InitState.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">wip-mobile/ducks</a> InitState.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/6</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29</td><td class="line-coverage quiet"><span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const initState = {
    problem : [], // for all problem in database
    newProblem : {
        topic: "",
        problem_type_id: 0,
        description: "",
        report_id: null,
        priority_id: 0
    }, // for report problem any input
    categoryProblem : [],
    priority : [],
    timetable: [],
    notification: []
}
&nbsp;
const INIT_STATE = "INIT_STATE"
&nbsp;
export default function <span class="fstat-no" title="function not covered" >reducer </span>(state = <span class="branch-0 cbranch-no" title="branch not covered" >initState,</span> action) {
<span class="cstat-no" title="statement not covered" >    switch (action.type) {</span>
        case INIT_STATE:
<span class="cstat-no" title="statement not covered" >            return initState</span>
        default:
<span class="cstat-no" title="statement not covered" >            return state</span>
    }
}
&nbsp;
export const SET_INIT = <span class="fstat-no" title="function not covered" >()</span> =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
    type: INIT_STATE
})</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Apr 06 2018 23:49:22 GMT+0700 (SE Asia Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
