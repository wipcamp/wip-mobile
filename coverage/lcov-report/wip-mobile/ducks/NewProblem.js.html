<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for wip-mobile/ducks/NewProblem.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">wip-mobile/ducks</a> NewProblem.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/22</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">16.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.82% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/17</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { initState } from './InitState'
&nbsp;
const SET_NEW_TOPIC = "SET_NEW_TOPIC"
const SET_NEW_TYPE = "SET_NEW_TYPE"
const SET_NEW_DESCRIPTION = "SET_NEW_DESCRIPTION"
const SET_NEW_REPORTID = "SET_NEW_REPORTID"
const SET_NEW_PRIORITY = "SET_NEW_PRIORITY"
&nbsp;
export default function <span class="fstat-no" title="function not covered" >reducer </span>(state = <span class="branch-0 cbranch-no" title="branch not covered" >initState,</span> action) {
<span class="cstat-no" title="statement not covered" >    switch (action.type) {</span>
        case SET_NEW_TOPIC : 
<span class="cstat-no" title="statement not covered" >            return {</span>
                newProblem : {
                    ...state.newProblem,
                    topic : action.topic
                }
            }
        case SET_NEW_TYPE :
<span class="cstat-no" title="statement not covered" >            return {</span>
                newProblem : {
                    ...state.newProblem,
                    problem_type_id : action.category
                }
            }
        case SET_NEW_DESCRIPTION : 
<span class="cstat-no" title="statement not covered" >            return {</span>
                newProblem : {
                    ...state.newProblem,
                    description : action.description
                }
            }
        case SET_NEW_REPORTID : 
<span class="cstat-no" title="statement not covered" >            return {</span>
                newProblem : {
                    ...state.newProblem,
                    report_id : action.id
                }
            }
        case SET_NEW_PRIORITY :
<span class="cstat-no" title="statement not covered" >            return {</span>
                newProblem : {
                    ...state.newProblem,
                    priority_id : action.id
                }
            }
        default : 
<span class="cstat-no" title="statement not covered" >            return state</span>
    }
}
&nbsp;
export const setNewProblemTopic = topic =&gt; ({
    type : SET_NEW_TOPIC,
    topic
})
&nbsp;
export const setNewProblemType = <span class="fstat-no" title="function not covered" >ca</span>tegory =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
    type : SET_NEW_TYPE,
    category
})
&nbsp;
export const setNewProblemDescription = <span class="fstat-no" title="function not covered" >de</span>scription =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
    type : SET_NEW_DESCRIPTION,
    description
})
&nbsp;
export const setNewProblemReportId = <span class="fstat-no" title="function not covered" >id</span> =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
    type : SET_NEW_REPORTID,
    id
})
&nbsp;
export const setNewProblemPriorityId = <span class="fstat-no" title="function not covered" >id</span> =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
    type : SET_NEW_PRIORITY,
    id
})</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Apr 06 2018 23:49:22 GMT+0700 (SE Asia Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
