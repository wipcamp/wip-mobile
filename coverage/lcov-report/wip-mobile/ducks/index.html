<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for wip-mobile/ducks</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> wip-mobile/ducks
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.11% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>32/76</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/27</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.39% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">47.62% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>30/63</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="CategoryProblem.js"><a href="CategoryProblem.js.html">CategoryProblem.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	</tr>

<tr>
	<td class="file low" data-value="InitState.js"><a href="InitState.js.html">InitState.js</a></td>
	<td data-value="42.86" class="pic low"><div class="chart"><div class="cover-fill" style="width: 42%;"></div><div class="cover-empty" style="width:58%;"></div></div></td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	</tr>

<tr>
	<td class="file medium" data-value="NewProblem.js"><a href="NewProblem.js.html">NewProblem.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="22" class="abs medium">11/22</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="58.82" class="pct medium">58.82%</td>
	<td data-value="17" class="abs medium">10/17</td>
	</tr>

<tr>
	<td class="file low" data-value="Priority.js"><a href="Priority.js.html">Priority.js</a></td>
	<td data-value="33.33" class="pic low"><div class="chart"><div class="cover-fill" style="width: 33%;"></div><div class="cover-empty" style="width:67%;"></div></div></td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	</tr>

<tr>
	<td class="file medium" data-value="ProblemTypeAPI.js"><a href="ProblemTypeAPI.js.html">ProblemTypeAPI.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	</tr>

<tr>
	<td class="file low" data-value="ReportProblem.js"><a href="ReportProblem.js.html">ReportProblem.js</a></td>
	<td data-value="32" class="pic low"><div class="chart"><div class="cover-fill" style="width: 32%;"></div><div class="cover-empty" style="width:68%;"></div></div></td>
	<td data-value="32" class="pct low">32%</td>
	<td data-value="25" class="abs low">8/25</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="38.1" class="pct low">38.1%</td>
	<td data-value="21" class="abs low">8/21</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Apr 06 2018 23:49:22 GMT+0700 (SE Asia Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
