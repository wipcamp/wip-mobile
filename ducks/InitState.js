export const initState = {
    problem : [], // for all problem in database
    newProblem : {
        topic: "",
        problem_type_id: 0,
        description: "",
        report_id: null,
        priority_id: 0
    }, // for report problem any input
    categoryProblem : [],
    priority : [],
    timetable: [],
    notification: []
}

const INIT_STATE = "INIT_STATE"

export default function reducer (state = initState, action) {
    switch (action.type) {
        case INIT_STATE:
            return initState
        default:
            return state
    }
}

export const SET_INIT = () => ({
    type: INIT_STATE
})