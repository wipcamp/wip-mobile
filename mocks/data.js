export const newProblem = {
    topic: "Test new problem",
    problem_type_id: 1,
    description: "Test new problem",
    report_id: 1
}

export const problem = [
    {
        id: 1,
        topic: "Test Problem 1",
        description: "Test Description 1",
        problem_type_id: 1,
        report_id: 1,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:01",
        updated_at: "2018-02-02 17:50:01"
    },
    {
        id: 2,
        topic: "Test Problem 2",
        description: "Test Description 2",
        problem_type_id: 2,
        report_id: 2,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:02",
        updated_at: "2018-02-02 17:50:02"
    }
]

export const categoryProblem = [
    {
        value: 1,
        label: "Test 1"
    },
    {
        value: 2,
        label: "Test 2"
    }
]

export const filter = 0

export const filterView = [
    {
        value: 0,
        label: "ทั้งหมด"
    },
    {
        label: 1,
        value: "Test 1"
    },
    {
        label: 2,
        value: "Test 2"
    }
]