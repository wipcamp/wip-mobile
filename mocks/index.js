const mockComponent = require.requireActual('react-native/jest/mockComponent')

module.exports = {
    initAll: () => {
      jest.mock('ScrollView', () => {
        const RealScrollView = require.requireActual('ScrollView')
        const ScrollView = mockComponent('ScrollView')
        ScrollView.prototype.render = RealScrollView.prototype.render
        return ScrollView
      })

      jest.mock('Dimensions', () => {
        const Dimensions = {
          get: jest.fn().mockReturnValue({width: 100, height:100})
        }
        return Dimensions
      })
    }
  }