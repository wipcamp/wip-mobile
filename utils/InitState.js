const initState = {
    filter : "ALL", 
    problem : [], // for all problem in database
    newProblem : {
        topic: null,
        problemTypeId: null,
        description: null,
        isSolve: false,
        reportId: null
    } // for report problem any input
}

export default initState