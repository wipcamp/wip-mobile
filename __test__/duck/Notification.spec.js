/**
 * @jest-environment node
 */

import reducer, {
    addNotification,
    reverseNotification,
    resetNotification
} from '../../ducks/Notification'

import { initState } from '../../ducks/InitState'
import { notification } from '../../mocks/data'

describe('action', () => {
    it('addNotification', () => {
        expect(addNotification(notification[0])).toMatchSnapshot()
    })

    it('reverseNotification', () => {
        expect(reverseNotification()).toMatchSnapshot()
    })

    it('resetNotification', () => {
        expect(resetNotification()).toMatchSnapshot()
    })
})

describe('reducer', () => {
    it('ADD_NOTIFICATION', () => {
        expect(reducer(initState, addNotification(notification[0]))).toMatchSnapshot()
    })

    it('REVERSE_NOTIFICATION', () => {
        expect(reducer(initState, reverseNotification())).toMatchSnapshot()
    })

    it('RESET_NOTIFICATION', () => {
        expect(reducer(initState, resetNotification())).toMatchSnapshot()
    })

    it('default', () => {
        expect(reducer(initState, {
            type: 'default'
        })).toMatchSnapshot()
    })
})