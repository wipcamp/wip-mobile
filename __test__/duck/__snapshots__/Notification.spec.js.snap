// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action addNotification 1`] = `
Object {
  "notification": Object {
    "body": "Test body 1",
    "created_at": "2018-02-02 17:50:01",
    "data": "{'table':'problems','id':1}",
    "id": 1,
    "title": "Test noti 1",
    "updated_at": "2018-02-02 17:50:01",
    "user_id": 1,
  },
  "type": "ADD_NOTIFICATION",
}
`;

exports[`action resetNotification 1`] = `
Object {
  "type": "RESET_NOTIFICATION",
}
`;

exports[`action reverseNotification 1`] = `
Object {
  "type": "REVERSE_NOTIFICATION",
}
`;

exports[`reducer ADD_NOTIFICATION 1`] = `
Object {
  "notification": Array [
    Object {
      "body": "Test body 1",
      "created_at": "2018-02-02 17:50:01",
      "data": "{'table':'problems','id':1}",
      "id": 1,
      "title": "Test noti 1",
      "updated_at": "2018-02-02 17:50:01",
      "user_id": 1,
    },
  ],
}
`;

exports[`reducer RESET_NOTIFICATION 1`] = `
Object {
  "notification": Array [],
}
`;

exports[`reducer REVERSE_NOTIFICATION 1`] = `
Object {
  "notification": Array [],
}
`;

exports[`reducer default 1`] = `
Object {
  "categoryProblem": Array [],
  "newProblem": Object {
    "description": "",
    "priority_id": 0,
    "problem_type_id": 0,
    "report_id": null,
    "topic": "",
  },
  "notification": Array [],
  "priority": Array [],
  "problem": Array [],
  "timetable": Array [],
}
`;
